src/main/java/in/projecteka/consentmanager/user/UserConfiguration.java

```diff
@@ -57,27 +63,46 @@ public IdentityServiceClient keycloakClient(WebClient.Builder builder,
 
     @Bean
     public SignUpService authenticatorService(JWTProperties jwtProperties,
-                                              LoadingCache<String, Optional<String>> sessionCache,
-                                              LoadingCache<String, Optional<String>> secondSessionCache,
+                                              CacheAdapter<String, String> sessionCache,
+                                              CacheAdapter<String, String> secondSessionCache,
                                               UserServiceProperties userServiceProperties) {
         return new SignUpService(jwtProperties,
                 sessionCache,
                 secondSessionCache,
                 userServiceProperties.getUserCreationTokenValidity());
     }
 
+    @Profile("default")
     @Bean({"unverifiedSessions", "verifiedSessions"})
-    public LoadingCache<String, Optional<String>> createSessionCache() {
+    public CacheAdapter<String, String> createLoadingCacheAdapter(LoadingCache<String,String> cache) {
+       return new LoadingCacheAdapter(cache);
+    }
+
+    @Bean
+    @Profile("default")
+    public LoadingCache<String, String> createSessionCache() {
         return CacheBuilder
                 .newBuilder()
                 .expireAfterWrite(5, TimeUnit.MINUTES)
-                .build(new CacheLoader<>() {
-                    public Optional<String> load(String key) {
-                        return Optional.empty();
+                .build(new CacheLoader<String, String>() {
+                    public String load(String key) {
+                        return "";
```

Comment:
return null

Reply:
Quoting from its doc
`   ... 
* <p><b>Warning:</b> as with {@link CacheLoader#load}, {@code loader} <b>must not</b> return
   * {@code null}; it may either return a non-null value or throw an exception.
...
`